packages/language-core/src/plugins/vue-tsx.ts

```diff
@@ -160,7 +160,7 @@ function createTsx(fileName: string, _sfc: Sfc, { vueCompilerOptions, compilerOp
 
 		for (const style of _sfc.styles) {
 			const option = vueCompilerOptions.experimentalResolveStyleCssClasses;
-			if ((option === 'always' || option === 'scoped') && style.scoped) {
+			if (option === 'always' || (option === 'scoped' && style.scoped)) {
```

Comment:
Just for reference [some lines above the code](https://github.com/vuejs/language-tools/pull/3689/files#diff-35e6594820018d2d5b7d52fc2dd8a2359445d24b2d54d5ae49cf1eaf742156f3R150) was initially correct